pkg/balancer-js/src/pool-weighted/normalizedWeights.ts

```diff
@@ -1,12 +1,22 @@
 import { BigNumber, BigNumberish } from '@ethersproject/bignumber';
 import { Zero, WeiPerEther as ONE } from '@ethersproject/constants';
 
+// Should match MAX_WEIGHTED_TOKENS from v2-helpers/constants
+// Including would introduce a dependency
+const MaxWeightedTokens = 100;
+
 /**
  * Normalize an array of token weights to ensure they sum to `1e18`
  * @param weights - an array of token weights to be normalized
  * @returns an equivalent set of normalized weights
  */
 export function toNormalizedWeights(weights: BigNumber[]): BigNumber[] {
+  // When the number is exactly equal to the max, normalizing with common inputs
+  // leads to a value < 0.01, which reverts. In this case fill in the weights exactly.
+  if (weights.length == MaxWeightedTokens) {
+    return Array(MaxWeightedTokens).fill(ONE.div(MaxWeightedTokens));
+  }
+
```

Comment:
This feels quite hacky. If we honestly think a pool with e.g. 90 tokens might exist, I'd consider allowing for smaller weights (maybe up to 0.1%?). It should be easy to review our arithmetic analysis to understand how this might change the limits (max join/exit ratio in join/exitswaps). cc @dmf7z

Reply:
Well, I mean the typescript convenience function that creates slightly different but similar weights by calling toNormalizedWeights(range(10000, 100100)) will have some < 0.01.

You mean change the Math to allow < 0.01? We did an analysis on that (see the limits Notion doc), and it seemed like 100 was a good limit, and I don't think more than 50ish will be practical.

I asked the main client waiting for this - they said 20 was the flagship, and they couldn't imagine doing more than 40. (Higher effective counts should just use composition.)

Reply:
I hear Nico's point but honestly I'm pretty content with this. I concur that it is very unlikely we ever see close to 100 tokens.